File: clients/csharp/src/Swagger/Client/Model/position.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;

namespace Swagger.Client.Model {
  public class Position {
    public double? account { get; set; }

    public string symbol { get; set; }

    public string underlying { get; set; }

    public string quoteCurrency { get; set; }

    public string currency { get; set; }

    public double? commission { get; set; }

    public double? prevRealisedPnl { get; set; }

    public double? prevUnrealisedPnl { get; set; }

    public double? prevClosePrice { get; set; }

    public double? realisedCost { get; set; }

    public double? realisedPnl { get; set; }

    public DateTime? openingTimestamp { get; set; }

    public double? openingQty { get; set; }

    public double? openingCost { get; set; }

    public double? openingComm { get; set; }

    public double? openOrderBuyQty { get; set; }

    public double? openOrderBuyCost { get; set; }

    public double? openOrderSellQty { get; set; }

    public double? openOrderSellCost { get; set; }

    public double? execBuyQty { get; set; }

    public double? execBuyCost { get; set; }

    public double? execSellQty { get; set; }

    public double? execSellCost { get; set; }

    public double? execQty { get; set; }

    public double? execCost { get; set; }

    public double? execComm { get; set; }

    public DateTime? currentTimestamp { get; set; }

    public double? currentQty { get; set; }

    public double? currentCost { get; set; }

    public double? currentComm { get; set; }

    public double? unrealisedCost { get; set; }

    public double? grossOpenCost { get; set; }

    public double? grossExecCost { get; set; }

    public double? lastPrice { get; set; }

    public double? lastValue { get; set; }

    public double? homeNotional { get; set; }

    public double? foreignNotional { get; set; }

    public double? initMargin { get; set; }

    public double? maintMargin { get; set; }

    public double? sessionMargin { get; set; }

    public double? targetExcessMargin { get; set; }

    public double? varMargin { get; set; }

    public double? unrealisedPnl { get; set; }

    public double? unrealisedPnlPcnt { get; set; }

    public double? simpleQty { get; set; }

    public double? simpleCost { get; set; }

    public double? simpleValue { get; set; }

    public double? simplePnl { get; set; }

    public double? simplePnlPcnt { get; set; }

    public double? avgEntryPrice { get; set; }

    public double? breakEvenPrice { get; set; }

    public double? marginCallPrice { get; set; }

    public double? liquidationPrice { get; set; }

    public double? bankruptPrice { get; set; }

    public DateTime? timestamp { get; set; }

    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class Position {\n");
      sb.Append("  account: ").Append(account).Append("\n");
      sb.Append("  symbol: ").Append(symbol).Append("\n");
      sb.Append("  underlying: ").Append(underlying).Append("\n");
      sb.Append("  quoteCurrency: ").Append(quoteCurrency).Append("\n");
      sb.Append("  currency: ").Append(currency).Append("\n");
      sb.Append("  commission: ").Append(commission).Append("\n");
      sb.Append("  prevRealisedPnl: ").Append(prevRealisedPnl).Append("\n");
      sb.Append("  prevUnrealisedPnl: ").Append(prevUnrealisedPnl).Append("\n");
      sb.Append("  prevClosePrice: ").Append(prevClosePrice).Append("\n");
      sb.Append("  realisedCost: ").Append(realisedCost).Append("\n");
      sb.Append("  realisedPnl: ").Append(realisedPnl).Append("\n");
      sb.Append("  openingTimestamp: ").Append(openingTimestamp).Append("\n");
      sb.Append("  openingQty: ").Append(openingQty).Append("\n");
      sb.Append("  openingCost: ").Append(openingCost).Append("\n");
      sb.Append("  openingComm: ").Append(openingComm).Append("\n");
      sb.Append("  openOrderBuyQty: ").Append(openOrderBuyQty).Append("\n");
      sb.Append("  openOrderBuyCost: ").Append(openOrderBuyCost).Append("\n");
      sb.Append("  openOrderSellQty: ").Append(openOrderSellQty).Append("\n");
      sb.Append("  openOrderSellCost: ").Append(openOrderSellCost).Append("\n");
      sb.Append("  execBuyQty: ").Append(execBuyQty).Append("\n");
      sb.Append("  execBuyCost: ").Append(execBuyCost).Append("\n");
      sb.Append("  execSellQty: ").Append(execSellQty).Append("\n");
      sb.Append("  execSellCost: ").Append(execSellCost).Append("\n");
      sb.Append("  execQty: ").Append(execQty).Append("\n");
      sb.Append("  execCost: ").Append(execCost).Append("\n");
      sb.Append("  execComm: ").Append(execComm).Append("\n");
      sb.Append("  currentTimestamp: ").Append(currentTimestamp).Append("\n");
      sb.Append("  currentQty: ").Append(currentQty).Append("\n");
      sb.Append("  currentCost: ").Append(currentCost).Append("\n");
      sb.Append("  currentComm: ").Append(currentComm).Append("\n");
      sb.Append("  unrealisedCost: ").Append(unrealisedCost).Append("\n");
      sb.Append("  grossOpenCost: ").Append(grossOpenCost).Append("\n");
      sb.Append("  grossExecCost: ").Append(grossExecCost).Append("\n");
      sb.Append("  lastPrice: ").Append(lastPrice).Append("\n");
      sb.Append("  lastValue: ").Append(lastValue).Append("\n");
      sb.Append("  homeNotional: ").Append(homeNotional).Append("\n");
      sb.Append("  foreignNotional: ").Append(foreignNotional).Append("\n");
      sb.Append("  initMargin: ").Append(initMargin).Append("\n");
      sb.Append("  maintMargin: ").Append(maintMargin).Append("\n");
      sb.Append("  sessionMargin: ").Append(sessionMargin).Append("\n");
      sb.Append("  targetExcessMargin: ").Append(targetExcessMargin).Append("\n");
      sb.Append("  varMargin: ").Append(varMargin).Append("\n");
      sb.Append("  unrealisedPnl: ").Append(unrealisedPnl).Append("\n");
      sb.Append("  unrealisedPnlPcnt: ").Append(unrealisedPnlPcnt).Append("\n");
      sb.Append("  simpleQty: ").Append(simpleQty).Append("\n");
      sb.Append("  simpleCost: ").Append(simpleCost).Append("\n");
      sb.Append("  simpleValue: ").Append(simpleValue).Append("\n");
      sb.Append("  simplePnl: ").Append(simplePnl).Append("\n");
      sb.Append("  simplePnlPcnt: ").Append(simplePnlPcnt).Append("\n");
      sb.Append("  avgEntryPrice: ").Append(avgEntryPrice).Append("\n");
      sb.Append("  breakEvenPrice: ").Append(breakEvenPrice).Append("\n");
      sb.Append("  marginCallPrice: ").Append(marginCallPrice).Append("\n");
      sb.Append("  liquidationPrice: ").Append(liquidationPrice).Append("\n");
      sb.Append("  bankruptPrice: ").Append(bankruptPrice).Append("\n");
      sb.Append("  timestamp: ").Append(timestamp).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }
  }
  }

File: clients/dart/lib/model/position.dart
part of api;


@Entity()
class Position {
  
  @Property(name: 'account')
  Number account = null;
  

  @Property(name: 'symbol')
  String symbol = null;
  

  @Property(name: 'currency')
  String currency = null;
  

  @Property(name: 'underlying')
  String underlying = null;
  

  @Property(name: 'quoteCurrency')
  String quoteCurrency = null;
  

  @Property(name: 'commission')
  double commission = 0.0;
  

  @Property(name: 'initMarginReq')
  double initMarginReq = 0.0;
  

  @Property(name: 'maintMarginReq')
  double maintMarginReq = 0.0;
  

  @Property(name: 'riskLimit')
  Number riskLimit = null;
  

  @Property(name: 'leverage')
  double leverage = 0.0;
  

  @Property(name: 'crossMargin')
  bool crossMargin = null;
  

  @Property(name: 'deleveragePercentile')
  double deleveragePercentile = 0.0;
  

  @Property(name: 'rebalancedPnl')
  Number rebalancedPnl = null;
  

  @Property(name: 'prevRealisedPnl')
  Number prevRealisedPnl = null;
  

  @Property(name: 'prevUnrealisedPnl')
  Number prevUnrealisedPnl = null;
  

  @Property(name: 'prevClosePrice')
  double prevClosePrice = 0.0;
  

  @Property(name: 'openingTimestamp')
  DateTime openingTimestamp = null;
  

  @Property(name: 'openingQty')
  Number openingQty = null;
  

  @Property(name: 'openingCost')
  Number openingCost = null;
  

  @Property(name: 'openingComm')
  Number openingComm = null;
  

  @Property(name: 'openOrderBuyQty')
  Number openOrderBuyQty = null;
  

  @Property(name: 'openOrderBuyCost')
  Number openOrderBuyCost = null;
  

  @Property(name: 'openOrderBuyPremium')
  Number openOrderBuyPremium = null;
  

  @Property(name: 'openOrderSellQty')
  Number openOrderSellQty = null;
  

  @Property(name: 'openOrderSellCost')
  Number openOrderSellCost = null;
  

  @Property(name: 'openOrderSellPremium')
  Number openOrderSellPremium = null;
  

  @Property(name: 'execBuyQty')
  Number execBuyQty = null;
  

  @Property(name: 'execBuyCost')
  Number execBuyCost = null;
  

  @Property(name: 'execSellQty')
  Number execSellQty = null;
  

  @Property(name: 'execSellCost')
  Number execSellCost = null;
  

  @Property(name: 'execQty')
  Number execQty = null;
  

  @Property(name: 'execCost')
  Number execCost = null;
  

  @Property(name: 'execComm')
  Number execComm = null;
  

  @Property(name: 'currentTimestamp')
  DateTime currentTimestamp = null;
  

  @Property(name: 'currentQty')
  Number currentQty = null;
  

  @Property(name: 'currentCost')
  Number currentCost = null;
  

  @Property(name: 'currentComm')
  Number currentComm = null;
  

  @Property(name: 'realisedCost')
  Number realisedCost = null;
  

  @Property(name: 'unrealisedCost')
  Number unrealisedCost = null;
  

  @Property(name: 'grossOpenCost')
  Number grossOpenCost = null;
  

  @Property(name: 'grossOpenPremium')
  Number grossOpenPremium = null;
  

  @Property(name: 'grossExecCost')
  Number grossExecCost = null;
  

  @Property(name: 'isOpen')
  bool isOpen = null;
  

  @Property(name: 'markPrice')
  double markPrice = 0.0;
  

  @Property(name: 'markValue')
  Number markValue = null;
  

  @Property(name: 'riskValue')
  Number riskValue = null;
  

  @Property(name: 'homeNotional')
  double homeNotional = 0.0;
  

  @Property(name: 'foreignNotional')
  double foreignNotional = 0.0;
  

  @Property(name: 'posState')
  String posState = null;
  

  @Property(name: 'posCost')
  Number posCost = null;
  

  @Property(name: 'posCost2')
  Number posCost2 = null;
  

  @Property(name: 'posCross')
  Number posCross = null;
  

  @Property(name: 'posInit')
  Number posInit = null;
  

  @Property(name: 'posComm')
  Number posComm = null;
  

  @Property(name: 'posLoss')
  Number posLoss = null;
  

  @Property(name: 'posMargin')
  Number posMargin = null;
  

  @Property(name: 'posMaint')
  Number posMaint = null;
  

  @Property(name: 'posAllowance')
  Number posAllowance = null;
  

  @Property(name: 'taxableMargin')
  Number taxableMargin = null;
  

  @Property(name: 'initMargin')
  Number initMargin = null;
  

  @Property(name: 'maintMargin')
  Number maintMargin = null;
  

  @Property(name: 'sessionMargin')
  Number sessionMargin = null;
  

  @Property(name: 'targetExcessMargin')
  Number targetExcessMargin = null;
  

  @Property(name: 'varMargin')
  Number varMargin = null;
  

  @Property(name: 'realisedGrossPnl')
  Number realisedGrossPnl = null;
  

  @Property(name: 'realisedTax')
  Number realisedTax = null;
  

  @Property(name: 'realisedPnl')
  Number realisedPnl = null;
  

  @Property(name: 'unrealisedGrossPnl')
  Number unrealisedGrossPnl = null;
  

  @Property(name: 'longBankrupt')
  Number longBankrupt = null;
  

  @Property(name: 'shortBankrupt')
  Number shortBankrupt = null;
  

  @Property(name: 'taxBase')
  Number taxBase = null;
  

  @Property(name: 'indicativeTaxRate')
  double indicativeTaxRate = 0.0;
  

  @Property(name: 'indicativeTax')
  Number indicativeTax = null;
  

  @Property(name: 'unrealisedTax')
  Number unrealisedTax = null;
  

  @Property(name: 'unrealisedPnl')
  Number unrealisedPnl = null;
  

  @Property(name: 'unrealisedPnlPcnt')
  double unrealisedPnlPcnt = 0.0;
  

  @Property(name: 'unrealisedRoePcnt')
  double unrealisedRoePcnt = 0.0;
  

  @Property(name: 'simpleQty')
  double simpleQty = 0.0;
  

  @Property(name: 'simpleCost')
  double simpleCost = 0.0;
  

  @Property(name: 'simpleValue')
  double simpleValue = 0.0;
  

  @Property(name: 'simplePnl')
  double simplePnl = 0.0;
  

  @Property(name: 'simplePnlPcnt')
  double simplePnlPcnt = 0.0;
  

  @Property(name: 'avgCostPrice')
  double avgCostPrice = 0.0;
  

  @Property(name: 'avgEntryPrice')
  double avgEntryPrice = 0.0;
  

  @Property(name: 'breakEvenPrice')
  double breakEvenPrice = 0.0;
  

  @Property(name: 'marginCallPrice')
  double marginCallPrice = 0.0;
  

  @Property(name: 'liquidationPrice')
  double liquidationPrice = 0.0;
  

  @Property(name: 'bankruptPrice')
  double bankruptPrice = 0.0;
  

  @Property(name: 'timestamp')
  DateTime timestamp = null;
  

  @Property(name: 'lastPrice')
  double lastPrice = 0.0;
  

  @Property(name: 'lastValue')
  Number lastValue = null;
  
  Position();

  @override
  String toString()  {
    return 'Position[account=$account, symbol=$symbol, currency=$currency, underlying=$underlying, quoteCurrency=$quoteCurrency, commission=$commission, initMarginReq=$initMarginReq, maintMarginReq=$maintMarginReq, riskLimit=$riskLimit, leverage=$leverage, crossMargin=$crossMargin, deleveragePercentile=$deleveragePercentile, rebalancedPnl=$rebalancedPnl, prevRealisedPnl=$prevRealisedPnl, prevUnrealisedPnl=$prevUnrealisedPnl, prevClosePrice=$prevClosePrice, openingTimestamp=$openingTimestamp, openingQty=$openingQty, openingCost=$openingCost, openingComm=$openingComm, openOrderBuyQty=$openOrderBuyQty, openOrderBuyCost=$openOrderBuyCost, openOrderBuyPremium=$openOrderBuyPremium, openOrderSellQty=$openOrderSellQty, openOrderSellCost=$openOrderSellCost, openOrderSellPremium=$openOrderSellPremium, execBuyQty=$execBuyQty, execBuyCost=$execBuyCost, execSellQty=$execSellQty, execSellCost=$execSellCost, execQty=$execQty, execCost=$execCost, execComm=$execComm, currentTimestamp=$currentTimestamp, currentQty=$currentQty, currentCost=$currentCost, currentComm=$currentComm, realisedCost=$realisedCost, unrealisedCost=$unrealisedCost, grossOpenCost=$grossOpenCost, grossOpenPremium=$grossOpenPremium, grossExecCost=$grossExecCost, isOpen=$isOpen, markPrice=$markPrice, markValue=$markValue, riskValue=$riskValue, homeNotional=$homeNotional, foreignNotional=$foreignNotional, posState=$posState, posCost=$posCost, posCost2=$posCost2, posCross=$posCross, posInit=$posInit, posComm=$posComm, posLoss=$posLoss, posMargin=$posMargin, posMaint=$posMaint, posAllowance=$posAllowance, taxableMargin=$taxableMargin, initMargin=$initMargin, maintMargin=$maintMargin, sessionMargin=$sessionMargin, targetExcessMargin=$targetExcessMargin, varMargin=$varMargin, realisedGrossPnl=$realisedGrossPnl, realisedTax=$realisedTax, realisedPnl=$realisedPnl, unrealisedGrossPnl=$unrealisedGrossPnl, longBankrupt=$longBankrupt, shortBankrupt=$shortBankrupt, taxBase=$taxBase, indicativeTaxRate=$indicativeTaxRate, indicativeTax=$indicativeTax, unrealisedTax=$unrealisedTax, unrealisedPnl=$unrealisedPnl, unrealisedPnlPcnt=$unrealisedPnlPcnt, unrealisedRoePcnt=$unrealisedRoePcnt, simpleQty=$simpleQty, simpleCost=$simpleCost, simpleValue=$simpleValue, simplePnl=$simplePnl, simplePnlPcnt=$simplePnlPcnt, avgCostPrice=$avgCostPrice, avgEntryPrice=$avgEntryPrice, breakEvenPrice=$breakEvenPrice, marginCallPrice=$marginCallPrice, liquidationPrice=$liquidationPrice, bankruptPrice=$bankruptPrice, timestamp=$timestamp, lastPrice=$lastPrice, lastValue=$lastValue, ]';
  }

}

